src/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.cs

```diff
@@ -36,22 +35,23 @@ public partial class CompareAttribute : System.ComponentModel.DataAnnotations.Va
         public override string FormatErrorMessage(string name) { throw null; }
         protected override System.ComponentModel.DataAnnotations.ValidationResult IsValid(object value, System.ComponentModel.DataAnnotations.ValidationContext validationContext) { throw null; }
     }
-    [System.AttributeUsageAttribute((System.AttributeTargets)(384), AllowMultiple = false, Inherited = true)]
+    [System.AttributeUsageAttribute(System.AttributeTargets.Field | System.AttributeTargets.Property, AllowMultiple=false, Inherited=true)]
     public sealed partial class ConcurrencyCheckAttribute : System.Attribute
     {
         public ConcurrencyCheckAttribute() { }
     }
-    [System.AttributeUsageAttribute((System.AttributeTargets)(2432), AllowMultiple = false)]
+    [System.AttributeUsageAttribute(System.AttributeTargets.Field | System.AttributeTargets.Parameter | System.AttributeTargets.Property, AllowMultiple=false)]
     public sealed partial class CreditCardAttribute : System.ComponentModel.DataAnnotations.DataTypeAttribute
     {
-        public CreditCardAttribute() : base(default(System.ComponentModel.DataAnnotations.DataType)) { }
+        public CreditCardAttribute() : base (default(System.ComponentModel.DataAnnotations.DataType)) { }
         public override bool IsValid(object value) { throw null; }
     }
-    [System.AttributeUsageAttribute((System.AttributeTargets)(2500), AllowMultiple = true)]
+    [System.AttributeUsageAttribute(System.AttributeTargets.Class | System.AttributeTargets.Field | System.AttributeTargets.Method | System.AttributeTargets.Parameter | System.AttributeTargets.Property, AllowMultiple=true)]
     public sealed partial class CustomValidationAttribute : System.ComponentModel.DataAnnotations.ValidationAttribute
     {
         public CustomValidationAttribute(System.Type validatorType, string method) { }
         public string Method { get { throw null; } }
+        public override bool RequiresValidationContext { get { throw null; } }
```

Comment:
It looks like we missed adding this. Should it be added?
cc @lajones, @divega, @ajcvickers

Reply:
@ahsonkhan I still have no context on why member overrides on a derived type make a difference in the ref assembly, but I can confirm that this particular override has been there from before 1.0 RTM.

Reply:
> I still have no context on why member overrides on a derived type make a difference in the ref assembly, but I can confirm that this particular override has been there from before 1.0 RTM.

In practice it only matters in one specific case: an override on a type that can be derived from.  In general compilers emit virtual calls to members, and as such it doesn't matter whether a method is overridden in the ref assembly, because a call site will invoke the virtual method from the base virtually and it'll end up invoking the override if it's there.  However, there's one key case where a call isn't virtual: when a derived type uses base.Method() to invoke the base method.  For that, if you don't have the override in the ref assembly and someone derives from you and does base.Method(), it's likely your override will end up being skipped.

In this case, CustomValidationAttribute is sealed, so no one can derive from it, and in practice it doesn't really matter.

Reply:
Ah, that totally makes sense. Thanks for the explanation.